qp_klp/klp.py

```diff
@@ -154,6 +186,7 @@ def _update_current_message(msg):
             qsamples = {
                 s.replace(f'{qiita_id}.', '') for s in qclient.get(qurl)}
             sample_name_diff = sheet_samples - qsamples
+            sn_tid_map_by_project[project_name] = None
```

Comment:
I think you are missing a `if sn_tid_map_by_project[project_name] is not None` somewhere in the code ... no?

Reply:
The FailedSamplesRecord is separate from the prep files that seqpro generates. map_sample_names_to_tube_ids() is just there to modify the prep files after seqpro creates them. If we need to modify the sample names for failed_samples.html and the other html file, we can do that as a next step.